src/DataGenerator.cpp

```diff
@@ -0,0 +1,499 @@
+  }
+}
+
+std::vector<std::string> DataGenerator::getCriteriaIds(std::string fileName) {
+  pugi::xml_document doc;
+  std::string path = "../data/" + fileName;
+  if (!doc.load_file(path.c_str()))
+    throw std::invalid_argument("Cannot open xml file, please check path");
+
+  // creating vector object to store criteria ids
+  std::vector<std::string> crit_ids;
+
+  if (DataGenerator::getXmlFileType(fileName) == "model") {
+    // if we have a model xml
+    // going from root ie model then to its first child modelName and passing
+    // to the 5th child ie first criterion
+    pugi::xml_node node_model = doc.child("model");
+
+    for (pugi::xml_node_iterator it = node_model.begin();
+         it != node_model.end(); ++it) {
+      if (strcmp(it->name(), "modelName") != 0 &&
+          strcmp(it->name(), "criteria") != 0 &&
+          strcmp(it->name(), "categories") != 0 &&
+          strcmp(it->name(), "lambda") != 0) {
+        const char *s = it->name();
+        std::string str(s);
+        crit_ids.push_back(s);
+      }
+    }
+  } else {
+    // if we have a dataset xml
+    pugi::xml_node node_dataset = doc.child("dataset").child("alternative");
+    for (pugi::xml_node_iterator it = node_dataset.begin();
+         it != node_dataset.end(); ++it) {
+      if (strcmp(it->name(), "") != 0 & strcmp(it->name(), "assignment") != 0) {
+        const char *s = it->name();
+        std::string str(s);
+        crit_ids.push_back(s);
+      }
+    }
+  }
+  return crit_ids;
+}
+
+Criterion DataGenerator::getCriterion(std::string fileName,
+                                      std::string crit_id) {
+  float weight = 0;
+  pugi::xml_document doc;
+  std::string path = "../data/" + fileName;
+
+  if (!doc.load_file(path.c_str()))
+    throw std::invalid_argument("Cannot open xml file, please check path");
+
+  if (DataGenerator::getXmlFileType(fileName) == "dataset") {
+    // if we have a model xml
+    throw std::invalid_argument(
+        "Cannot find the criterion associated to its weights in xml file, "
+        "most likely have a xml model file");
+  } else {
+    // if we have a dataset xml
+    pugi::xml_node node_model = doc.child("model").child("modelName");
+
+    for (pugi::xml_node_iterator it = node_model.begin();
+         it != node_model.end(); ++it) {
+
+      if (it->child_value() == crit_id) {
+        // std::cout << "criterion : " << it->child_value() << std::endl;
+        pugi::xml_node criterion_node = node_model.child(it->name());
+
+        for (pugi::xml_node_iterator al_it = criterion_node.begin();
+             al_it != criterion_node.end(); ++al_it) {
+
+          // std::cout << "crit : " << al_it->child_value() << std::endl;
+          if (strcmp(al_it->child_value(), "weight") == 0) {
+            weight = atof(al_it->child_value());
+          }
+        }
+        break;
+      }
+    }
+  }
+  return Criterion(crit_id, weight);
+}
+
+int DataGenerator::getAlternativeAssignment(std::string fileName,
```

Comment:
We probably need to handle the case of not finding alt_id here as well

Reply:
Done as well, created new tests